api-lib/sendSocialMessage.ts

```diff
@@ -176,16 +174,105 @@ export async function sendSocialMessage({
       chat_id: channelId,
       text: msg,
     };
-    const res = await fetch(`${TELEGRAM_BOT_BASE_URL}/sendMessage`, {
-      method: 'POST',
+
+    const updateTelegramBot = fetchUrl({
+      url: `${TELEGRAM_BOT_BASE_URL}/sendMessage`,
       body: JSON.stringify(telegramBotPost),
+      label: 'telegram-bot',
+      notifyOrg,
+      circleId,
+      channelId,
+    });
+    requests.push(updateTelegramBot);
+  }
+
+  const responses = await Promise.allSettled(requests);
+
+  const errors = responses.filter(isRejected);
+
+  if (errors.length > 0) {
+    const errorMessages = errors.map(err => {
+      if (err.reason instanceof Error) {
+        return err.reason.stack;
+      } else {
+        return String(err.reason);
+      }
+    });
+    console.error(`Error sending social messages: ${errorMessages.join('\n')}`);
+    throw new Error(errorMessages.join('\n'));
+  }
+}
+
+const isRejected = (
+  response: PromiseSettledResult<unknown>
+): response is PromiseRejectedResult => response.status === 'rejected';
+
+const fetchUrl = async ({
+  url,
+  headers,
+  body,
+  label,
+  notifyOrg,
+  channelId,
+  circleId,
+}: {
+  url: RequestInfo;
+  headers?: HeaderInit;
+  body: BodyInit;
+  label: string;
+  notifyOrg: boolean;
+  channelId?: string;
+  circleId: number;
+}) => {
+  try {
+    const updateWebhook = await fetch(url, {
+      method: 'POST',
+      body: body,
       headers: {
         'Content-Type': 'application/json',
+        ...headers,
       },
     });
+    if (!updateWebhook.ok) {
+      if (channelId) {
+        console.error(
+          `Error updating ${label}:`,
+          updateWebhook.status,
+          updateWebhook.statusText,
+          `, Circle Id: ${circleId}`,
+          `, Channel Id: ${channelId}`,
+          `, notifyOrg: ${notifyOrg}`
+        );
+      } else {
+        console.error(
+          `Error updating ${label}:`,
+          updateWebhook.status,
+          updateWebhook.statusText,
+          `, Circle Id: ${circleId}`,
+          `, notifyOrg: ${notifyOrg}`
+        );
+      }
 
-    if (!res.ok) {
-      throw new Error(JSON.stringify(await res.json()));
+      const json = await updateWebhook.json();
+      throw new Error(JSON.stringify(json));
     }
+  } catch (err: any) {
+    if (channelId) {
+      console.error(
```

Comment:
these two `console.error` lines are identical except for `channelId`. so it could just be one:
```js
console.error('...' + (channelId ? `Channel Id: ${channelId}` : ''));
```